src/Microsoft.OData.Core/Uri/ApplyClauseToStringBuilder.cs

```diff
@@ -148,6 +154,24 @@ private void Translate(ComputeTransformationNode transformation)
             }
         }
 
+        private void Translate(ExpandTransformationNode transformation)
+        {
+            ExpandedNavigationSelectItem expandedNavigation = transformation.ExpandClause.SelectedItems.Single() as ExpandedNavigationSelectItem;
+            AppendExpression(expandedNavigation.PathToNavigationProperty);
+            if (expandedNavigation.FilterOption != null || expandedNavigation.SelectAndExpand != null)
+            {
+                AppendComma(true);
+                if (expandedNavigation.FilterOption != null)
+                {
+                    query.Append(ExpressionConstants.SymbolEscapedSpace);
+                    query.Append(ExpressionConstants.KeywordFilter);
+                    query.Append(ExpressionConstants.SymbolOpenParen);
+                    AppendExpression(expandedNavigation.FilterOption.Expression);
+                    query.Append(ExpressionConstants.SymbolClosedParen);
+                }

```

Comment:
Please check this logic; it appears that, if expandedNavigation.FilterOption is null but expandedNavigation.SelectAndExpand is not null, we'll append a comma but nothing else. Otherwise, if expandedNavigation.FilterOption is non-null, we'll append the comma, followed by the parens expression and no trailing comma. Is this intended?

Reply:
Fixed. Also added support for nested expands (I missed that initially)